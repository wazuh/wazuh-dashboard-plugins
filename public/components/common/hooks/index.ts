/*
 * Wazuh app - React hooks
 * Copyright (C) 2015-2022 Wazuh, Inc.
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Find more information about this on the LICENSE file.
 */

export * from './use-filter-manager';
export * from './use-index-pattern';
export * from './use-kbn-loading-indicator';
export * from './use-query';
export * from './use-time-filter';
export * from './useWindowSize';
export * from './useUserPermissions';
export * from './useUserRoles';
export * from './useResfreshAngularDiscover';
export * from './useAllowedAgents';
export * from './useApiRequest';
export * from './use-app-config';
export * from './useRootScope';
export * from './use_async_action';
export * from './use_async_action_run_on_start';
export { useEsSearch } from './use-es-search';
export { useValueSuggestion, IValueSuggestion } from './use-value-suggestion';
