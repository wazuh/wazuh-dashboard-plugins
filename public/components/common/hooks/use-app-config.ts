/*
 * Wazuh app - React hook for app configuration
 * Copyright (C) 2015-2022 Wazuh, Inc.
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Find more information about this on the LICENSE file.
 */

import { AppRootState } from '../../../redux/types';
import { useSelector } from 'react-redux';

export const useAppConfig = () => {
  const appConfig = useSelector((state: AppRootState) => state.appConfig);
  return appConfig;
}
