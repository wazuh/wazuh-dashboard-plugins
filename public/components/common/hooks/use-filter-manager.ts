/*
 * Wazuh app - React hook for get Kibana filter manager
 * Copyright (C) 2015-2021 Wazuh, Inc.
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Find more information about this on the LICENSE file.
 */
import { getDataPlugin } from '../../../kibana-services';
import { useState, useEffect} from 'react';


export const useFilterManager = () => {
    const [filterManager, setFilterManager] = useState(getDataPlugin().query.filterManager);
    return filterManager;
}