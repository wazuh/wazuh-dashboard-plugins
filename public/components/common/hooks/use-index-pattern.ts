/*
 * Wazuh app - React hook for get wazuh-alert index pattern
 * Copyright (C) 2015-2022 Wazuh, Inc.
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Find more information about this on the LICENSE file.
 */
import { useState, useEffect} from 'react';
//@ts-ignore
import { AppState } from '../../../react-services/app-state';
import { IIndexPattern } from '../../../../../../src/plugins/data/public';
import { getDataPlugin } from '../../../kibana-services';

export const useIndexPattern = (): IIndexPattern | undefined => {
  const [indexPattern, setIndexPattern] = useState();
  useEffect(() => {
  const idIndexPattern = AppState.getCurrentPattern();
  getDataPlugin().indexPatterns.get(idIndexPattern)
    .then(setIndexPattern);
  }, []);
  return indexPattern;
}