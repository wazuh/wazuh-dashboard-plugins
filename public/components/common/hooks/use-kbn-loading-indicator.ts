/*
 * Wazuh app - React hook hidde or show the plugin platform loading indicator
 * Copyright (C) 2015-2022 Wazuh, Inc.
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Find more information about this on the LICENSE file.
 */
import { getHttp } from '../../../kibana-services';
import React, { useEffect, useState } from 'react';
import { BehaviorSubject } from 'rxjs';

export const useKbnLoadingIndicator = (): [
  boolean,
  React.Dispatch<React.SetStateAction<boolean>>,
  boolean
] => {
  const [loading, setLoading] = useState(false);
  const [flag, setFlag] = useState(false);
  const [visible, setVisible] = useState(0);

  const loadingCount$ = new BehaviorSubject(0);

  useEffect(() => {
    getHttp().addLoadingCountSource(loadingCount$);
    const { unsubscribe } = getHttp() 
      .getLoadingCount$()
      .subscribe((count) => {
        setVisible(count);
        !count && setFlag(false);
      });
    return unsubscribe;
  }, []);

  useEffect(() => {
    if (loading && visible <= 0) {
      loadingCount$.next(loadingCount$.value + 1);
      setFlag(true);
    }

    if (!loading && flag && visible > 0) {
      loadingCount$.next(loadingCount$.value - 1);
    }
  }, [visible, loading]);
  return [loading, setLoading, visible > 0];
};