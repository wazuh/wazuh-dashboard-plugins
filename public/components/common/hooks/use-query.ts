/*
 * Wazuh app - React hook for get query of Kibana searchBar
 * Copyright (C) 2015-2021 Wazuh, Inc.
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Find more information about this on the LICENSE file.
 */
import { getDataPlugin } from '../../../kibana-services';
import { useState, useEffect } from 'react';
import { ModulesHelper } from '../modules/modules-helper';

export function useQuery(): [{
  language: 'kuery' | 'lucene';
  query: string;
}, ((query: any) => void)] {
  const [query, setQuery] = useState({ language: 'kuery', query: '' });
  useEffect(() => {
    let subscription;
    ModulesHelper.getDiscoverScope()
      .then(scope => {
        setQuery(scope.state.query);
        subscription = scope.$watchCollection('fetchStatus',
        () => {setQuery(scope.state.query)});
      });
    return () => { subscription && subscription(); }
  }, []);
  const updateQuery = (query) => {
    ModulesHelper.getDiscoverScope()
      .then(scope => {
        scope.state.query = query;
      })
  }
  return [ query, updateQuery ];
}

export const useQueryManager = () => {
  return useState(getDataPlugin().query.queryString.getQuery());
}
