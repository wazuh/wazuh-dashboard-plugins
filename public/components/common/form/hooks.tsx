import { useState, useRef } from 'react';
import { isEqual } from 'lodash';
import { EpluginSettingType } from '../../../../common/constants';

function getValueFromEvent(event, type){
  return (getValueFromEventType[type] || getValueFromEventType.default)(event);
};

const getValueFromEventType = {
  [EpluginSettingType.switch] : (event: any) => event.target.checked,
  [EpluginSettingType.editor]: (value: any) => value,
  [EpluginSettingType.filepicker]: (value: any) => value,
  default: (event: any) => event.target.value,
};

export const useForm = (fields) => {
  const [formFields, setFormFields] = useState(Object.entries(fields).reduce((accum, [fieldKey, fieldConfiguration]) => ({
    ...accum,
    [fieldKey]: {
      currentValue: fieldConfiguration.initialValue,
      initialValue: fieldConfiguration.initialValue,
    }
  }), {}));

  const fieldRefs = useRef({});

  const enhanceFields = Object.entries(formFields).reduce((accum, [fieldKey, {currentValue: value, ...restFieldState}]) => ({
    ...accum,
    [fieldKey]: {
      ...fields[fieldKey],
      ...restFieldState,
      type: fields[fieldKey].type,
      value,
      changed: !isEqual(restFieldState.initialValue, value),
      error: fields[fieldKey]?.validate?.(value),
      setInputRef: (reference) => {fieldRefs.current[fieldKey] = reference},
      inputRef: fieldRefs.current[fieldKey],
      onChange: (event) => {
        const inputValue = getValueFromEvent(event, fields[fieldKey].type);
        const currentValue = fields[fieldKey]?.transformChangedInputValue?.(inputValue) ?? inputValue;
        setFormFields(state => ({
          ...state,
          [fieldKey]: {
            ...state[fieldKey],
            currentValue,
          }
        }))
      },
    }
  }), {});

  const changed = Object.fromEntries(
    Object.entries(enhanceFields).filter(([, {changed}]) => changed).map(([fieldKey, {value}]) => ([fieldKey, fields[fieldKey]?.transformChangedOutputValue?.(value) ?? value]))
  );

  const errors = Object.fromEntries(
    Object.entries(enhanceFields).filter(([, {error}]) => error).map(([fieldKey, {error}]) => ([fieldKey, error]))
  );

  function undoChanges(){
    setFormFields(state => Object.fromEntries(
      Object.entries(state).map(([fieldKey, fieldConfiguration]) => ([
        fieldKey,
        {
          ...fieldConfiguration,
          currentValue: fieldConfiguration.initialValue
        }
      ]))
    ));
  };

  function doChanges(){
    setFormFields(state => Object.fromEntries(
      Object.entries(state).map(([fieldKey, fieldConfiguration]) => ([
        fieldKey,
        {
          ...fieldConfiguration,
          initialValue: fieldConfiguration.currentValue
        }
      ]))
    ));
  };

  return {
    fields: enhanceFields,
    changed,
    errors,
    undoChanges,
    doChanges
  };
};
