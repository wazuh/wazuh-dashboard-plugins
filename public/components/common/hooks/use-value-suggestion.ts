/*
 * Wazuh app - React hook for getting value suggestions
 * Copyright (C) 2015-2022 Wazuh, Inc.
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Find more information about this on the LICENSE file.
 */

import React, { useEffect, useState } from 'react';
import { getDataPlugin } from '../../../kibana-services';
import { useIndexPattern } from '../hooks';
import { IFieldType, IIndexPattern } from 'src/plugins/data/public';
import {
  UI_ERROR_SEVERITIES,
  UIErrorLog,
  UIErrorSeverity,
  UILogLevel,
} from '../../../react-services/error-orchestrator/types';
import { UI_LOGGER_LEVELS } from '../../../../common/constants';
import { getErrorOrchestrator } from '../../../react-services/common-services';

export interface IValueSuggestion {
  suggestedValues: string[] | boolean[];
  isLoading: boolean;
  setQuery: React.Dispatch<React.SetStateAction<string>>;
}

interface BoolFilter {
  field: string;
  value: string;
}

export const useValueSuggestion = (
  filterField: string,
  boolFilterValue: BoolFilter = {
    field: '',
    value: '',
  },
  options?: string[],
  type: 'string' | 'boolean' = 'string'
): IValueSuggestion => {
  const [suggestedValues, setSuggestedValues] = useState<string[] | boolean[]>([]);
  const [query, setQuery] = useState<string>('');
  const [isLoading, setIsLoading] = useState(true);
  const data = getDataPlugin();
  const indexPattern = useIndexPattern();

  const getOptions = (): string[] => {
    return options?.filter((element) => element.toLowerCase().includes(query.toLowerCase())) || [];
  };

  const getValueSuggestions = async (field) => {
    const boolFilter =
      boolFilterValue.value !== ''
        ? [
            {
              term: {
                [boolFilterValue.field]: `${boolFilterValue.value}`,
              },
            },
          ]
        : [];
    return options
      ? getOptions()
      : await data.autocomplete.getValueSuggestions({
          query,
          indexPattern: indexPattern as IIndexPattern,
          field: { ...field, toSpec: (options) => field },
          boolFilter: boolFilter,
        });
  };

  useEffect(() => {
    if (indexPattern) {
      setIsLoading(true);
      (async () => {
        const field = {
          type: type,
          name: filterField,
          aggregatable: true,
        } as IFieldType;
        try {
          setSuggestedValues(await getValueSuggestions(field));
        } catch (error) {
          const options: UIErrorLog = {
            context: `${useValueSuggestion.name}.getValueSuggestions`,
            level: UI_LOGGER_LEVELS.ERROR as UILogLevel,
            severity: UI_ERROR_SEVERITIES.UI as UIErrorSeverity,
            error: {
              error,
              message: error.message || error,
              title: error.name,
            },
          };
          getErrorOrchestrator().handleError(options);
        } finally {
          setIsLoading(false);
        }
      })();
    }
  }, [indexPattern, query, filterField, type, boolFilterValue]);

  return { suggestedValues, isLoading, setQuery };
};
