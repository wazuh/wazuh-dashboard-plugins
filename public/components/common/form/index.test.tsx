import React from 'react';
import { render, screen } from '@testing-library/react';
import { InputForm } from './index';
import { useForm } from './hooks';

jest.mock('../../../../../../node_modules/@elastic/eui/lib/services/accessibility/html_id_generator', () => ({
	htmlIdGenerator: () => () => 'test-id',
}));

describe('[component] InputForm', () => {
	const optionsEditor = {editor: {language: 'json'}};
	const optionsFilepicker = {file: {type: 'image', extensions: ['.jpeg', '.jpg', '.png', '.svg']}};
	const optionsSelect = {select: [{text: 'Label1', value: 'value1'}, {text: 'Label2', value: 'value2'}]};
	const optionsSwitch = {switch: {values: {enabled: {label: 'Enabled', value: true}, disabled: {label: 'Disabled', value: false}}}};
	it.each`
		inputType       | value       | options
		${'editor'}     | ${'{}'}     | ${optionsEditor}
		${'filepicker'} | ${'{}'}     | ${optionsFilepicker}
		${'number'}     | ${4}        | ${undefined}
		${'select'}     | ${'value1'} | ${optionsSelect}
		${'switch'}     | ${true}     | ${optionsSwitch}
		${'text'}       | ${'test'}   | ${undefined}
		${'textarea'}   | ${'test'}   | ${undefined}
	`('Renders correctly to match the snapshot: Input: $inputType', ({ inputType, value, options }) => {
		const wrapper = render(
			<InputForm
				type={inputType}
				value={value}
				onChange={() => {}}
				options={options}
			/>
		);
		expect(wrapper.container).toMatchSnapshot();
	});

	it.each`
		inputType | initialValue | options | rest
		${'number'} | ${4} | ${{number: {min: 5}}} | ${{ validate: (value) => value > 3 ? undefined : 'Vaidation error: value is lower than 5'}}
		${'text'} | ${''} | ${undefined} | ${{ validate: (value) => value.length ? undefined : 'Validation error: string can not be empty' }}
		${'text'} | ${'test spaces'} | ${undefined} | ${{ validate: (value) => value.length ? undefined : 'Validation error: string can not contain spaces' }}
	`('Renders correctly to match the snapshot with validation errors. Input: $inputType', async ({ inputType, initialValue, options, rest }) => {
		const TestComponent = () => {
			const { fields: { [inputType]: field } } = useForm({ [inputType]: { initialValue, type: inputType, options, ...rest } });
			return (
				<InputForm
					label={'Test'}
					{...field}
				/>
			);
		};
		const wrapper = render(<TestComponent />);
		expect(wrapper.container).toMatchSnapshot();
	});
});

