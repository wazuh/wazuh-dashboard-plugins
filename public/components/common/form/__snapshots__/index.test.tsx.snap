// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[component] InputForm Renders correctly to match the snapshot with validation errors. Input: number 1`] = `
<div>
  <div
    class="euiFormRow euiFormRow--fullWidth"
    id="test-id-row"
  >
    <div
      class="euiFormRow__labelWrapper"
    >
      <label
        aria-invalid="false"
        class="euiFormLabel euiFormRow__label"
        for="test-id"
      >
        Test
      </label>
    </div>
    <div
      class="euiFormRow__fieldWrapper"
    >
      <div
        class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionRow"
      >
        <div
          class="euiFlexItem"
        >
          <div
            class="euiFormControlLayout euiFormControlLayout--fullWidth"
          >
            <div
              class="euiFormControlLayout__childrenWrapper"
            >
              <input
                class="euiFieldNumber euiFieldNumber--fullWidth"
                min="5"
                type="number"
                value="4"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`[component] InputForm Renders correctly to match the snapshot with validation errors. Input: text 1`] = `
<div>
  <div
    class="euiFormRow euiFormRow--fullWidth"
    id="test-id-row"
  >
    <div
      class="euiFormRow__labelWrapper"
    >
      <label
        aria-invalid="true"
        class="euiFormLabel euiFormRow__label euiFormLabel-isInvalid"
        for="test-id"
      >
        Test
      </label>
    </div>
    <div
      class="euiFormRow__fieldWrapper"
    >
      <div
        class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionRow"
      >
        <div
          class="euiFlexItem"
        >
          <div
            class="euiFormControlLayout euiFormControlLayout--fullWidth"
          >
            <div
              class="euiFormControlLayout__childrenWrapper"
            >
              <input
                class="euiFieldText euiFieldText--fullWidth"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        aria-live="polite"
        class="euiFormErrorText euiFormRow__text"
        id="test-id-error-0"
      >
        Validation error: string can not be empty
      </div>
    </div>
  </div>
</div>
`;

exports[`[component] InputForm Renders correctly to match the snapshot with validation errors. Input: text 2`] = `
<div>
  <div
    class="euiFormRow euiFormRow--fullWidth"
    id="test-id-row"
  >
    <div
      class="euiFormRow__labelWrapper"
    >
      <label
        aria-invalid="false"
        class="euiFormLabel euiFormRow__label"
        for="test-id"
      >
        Test
      </label>
    </div>
    <div
      class="euiFormRow__fieldWrapper"
    >
      <div
        class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionRow"
      >
        <div
          class="euiFlexItem"
        >
          <div
            class="euiFormControlLayout euiFormControlLayout--fullWidth"
          >
            <div
              class="euiFormControlLayout__childrenWrapper"
            >
              <input
                class="euiFieldText euiFieldText--fullWidth"
                type="text"
                value="test spaces"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`[component] InputForm Renders correctly to match the snapshot: Input: editor 1`] = `
<div>
  <div
    class="euiCodeEditorWrapper"
    data-test-subj="codeEditorContainer"
    style="width: 100%; height: 50px;"
  >
    <div
      class="euiCodeEditorKeyboardHint"
      data-test-subj="codeEditorHint"
      id="test-id"
      role="button"
      tabindex="0"
    >
      <p
        class="euiText"
      >
        Press Enter to start editing.
      </p>
      <p
        class="euiText"
      >
        When you're done, press Escape to stop editing.
      </p>
    </div>
    <div
      class=" ace_editor ace-tm"
      id="test-id"
      style="width: 100%; height: 50px;"
    >
      <textarea
        autocapitalize="off"
        autocorrect="off"
        class="ace_text-input"
        spellcheck="false"
        style="opacity: 0;"
        tabindex="-1"
        wrap="off"
      />
      <div
        aria-hidden="true"
        class="ace_gutter"
      >
        <div
          class="ace_layer ace_gutter-layer ace_folding-enabled"
        />
        <div
          class="ace_gutter-active-line"
        />
      </div>
      <div
        class="ace_scroller"
      >
        <div
          class="ace_content"
        >
          <div
            class="ace_layer ace_print-margin-layer"
          >
            <div
              class="ace_print-margin"
              style="left: 4px; visibility: visible;"
            />
          </div>
          <div
            class="ace_layer ace_marker-layer"
          />
          <div
            class="ace_layer ace_text-layer"
            style="padding: 0px 4px;"
          />
          <div
            class="ace_layer ace_marker-layer"
          />
          <div
            class="ace_layer ace_cursor-layer ace_hidden-cursors"
          >
            <div
              class="ace_cursor"
            />
          </div>
        </div>
      </div>
      <div
        class="ace_scrollbar ace_scrollbar-v"
        style="display: none; width: 20px;"
      >
        <div
          class="ace_scrollbar-inner"
          style="width: 20px;"
        />
      </div>
      <div
        class="ace_scrollbar ace_scrollbar-h"
        style="display: none; height: 20px;"
      >
        <div
          class="ace_scrollbar-inner"
          style="height: 20px;"
        />
      </div>
      <div
        style="height: auto; width: auto; top: 0px; left: 0px; visibility: hidden; position: absolute; white-space: pre; overflow: hidden;"
      >
        <div
          style="height: auto; width: auto; top: 0px; left: 0px; visibility: hidden; position: absolute; white-space: pre; overflow: visible;"
        />
        <div
          style="height: auto; width: auto; top: 0px; left: 0px; visibility: hidden; position: absolute; white-space: pre; overflow: visible;"
        >
          XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`[component] InputForm Renders correctly to match the snapshot: Input: filepicker 1`] = `
<div>
  <div
    class="euiFilePicker euiFilePicker--fullWidth"
  >
    <div
      class="euiFilePicker__wrap"
    >
      <input
        accept=".jpeg,.jpg,.png,.svg"
        aria-describedby="test-id"
        aria-label="Upload a file"
        class="euiFilePicker__input"
        type="file"
      />
      <div
        class="euiFilePicker__prompt"
        id="test-id"
      >
        <svg
          aria-hidden="true"
          class="euiIcon euiIcon--medium euiIcon-isLoading euiFilePicker__icon"
          focusable="false"
          height="16"
          role="img"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        />
        <div
          class="euiFilePicker__promptText"
        >
          Select or drag the file
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`[component] InputForm Renders correctly to match the snapshot: Input: number 1`] = `
<div>
  <div
    class="euiFormControlLayout euiFormControlLayout--fullWidth"
  >
    <div
      class="euiFormControlLayout__childrenWrapper"
    >
      <input
        class="euiFieldNumber euiFieldNumber--fullWidth"
        type="number"
        value="4"
      />
    </div>
  </div>
</div>
`;

exports[`[component] InputForm Renders correctly to match the snapshot: Input: select 1`] = `
<div>
  <div
    class="euiFormControlLayout"
  >
    <div
      class="euiFormControlLayout__childrenWrapper"
    >
      <select
        class="euiSelect"
      >
        <option
          value="value1"
        >
          Label1
        </option>
        <option
          value="value2"
        >
          Label2
        </option>
      </select>
      <div
        class="euiFormControlLayoutIcons euiFormControlLayoutIcons--right"
      >
        <span
          class="euiFormControlLayoutCustomIcon"
        >
          <svg
            aria-hidden="true"
            class="euiIcon euiIcon--medium euiIcon-isLoading euiFormControlLayoutCustomIcon__icon"
            focusable="false"
            height="16"
            role="img"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          />
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`[component] InputForm Renders correctly to match the snapshot: Input: switch 1`] = `
<div>
  <div
    class="euiSwitch"
  >
    <button
      aria-checked="true"
      aria-labelledby="test-id"
      class="euiSwitch__button"
      id="test-id"
      role="switch"
      type="button"
    >
      <span
        class="euiSwitch__body"
      >
        <span
          class="euiSwitch__thumb"
        />
        <span
          class="euiSwitch__track"
        >
          <svg
            aria-hidden="true"
            class="euiIcon euiIcon--medium euiIcon-isLoading euiSwitch__icon"
            focusable="false"
            height="16"
            role="img"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          />
          <svg
            aria-hidden="true"
            class="euiIcon euiIcon--medium euiIcon-isLoading euiSwitch__icon euiSwitch__icon--checked"
            focusable="false"
            height="16"
            role="img"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          />
        </span>
      </span>
    </button>
    <span
      class="euiSwitch__label"
      id="test-id"
    >
      Enabled
    </span>
  </div>
</div>
`;

exports[`[component] InputForm Renders correctly to match the snapshot: Input: text 1`] = `
<div>
  <div
    class="euiFormControlLayout euiFormControlLayout--fullWidth"
  >
    <div
      class="euiFormControlLayout__childrenWrapper"
    >
      <input
        class="euiFieldText euiFieldText--fullWidth"
        type="text"
        value="test"
      />
    </div>
  </div>
</div>
`;

exports[`[component] InputForm Renders correctly to match the snapshot: Input: textarea 1`] = `
<div>
  <textarea
    class="euiTextArea euiTextArea--resizeVertical euiTextArea--fullWidth"
    rows="6"
  >
    test
  </textarea>
</div>
`;
